### PR TITLE
Add ability for cPanel API requests to use API tokens.

### PR DESCRIPTION
**Pull request by FGasper**

### CODE REVIEW

**Comment by FGasper** [comment]
I’ll squash this once it’s past review.

File: lib/cPanel/PublicAPI.pm [hunk 1]
```diff
@@ -289,6 +289,17 @@ sub api_request {
             }
         }
 
+        if ($self->{'operating_mode'} eq 'accesshash') {
+            my $token_app = ($service eq 'whostmgr') ? 'whm' : $service;
```
Comment (bdraco) [inline lib/cPanel/PublicAPI.pm]: @FGasper  Shouldn't this be uppercase to maintain compat?

Comment (FGasper) [reply lib/cPanel/PublicAPI.pm]: Documentation gives it in lowercase: https://documentation.cpanel.net/display/DD/Guide+to+API+Authentication+-+API+Tokens+in+WHM

Comment (bdraco) [reply lib/cPanel/PublicAPI.pm]: Confirm lowercase works as well
 elsif ( Cpanel::App::is_whm() && $authtype =~ m{whm}i ) {

**Comment by bdraco** [comment]
Add ability for cPanel API requests to use API tokens

**Pull request merged**